src/components/Canvas.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import Node from './Node';
+import Edge from './Edge';
+
+interface IEdge {
+  from: number;
+  to: number;
+};
+
+interface ImappedNode {
+  label?: string;
+  id: number;
+  x: number;
+  y: number;
+}
+
+interface ImappedEdge {
```

Comment:
Nice, but I'd remove **I** since it's not common naming in TS (and looks too much C#-ish ;) )
https://www.typescriptlang.org/docs/handbook/interfaces.html